src/Benchmark/FlatteningMapper.cs

```diff
@@ -67,7 +67,7 @@ public class Customer
             public Address Address { get; set; }
             public Address HomeAddress { get; set; }
             public Address[] Addresses { get; set; }
-            public ICollection<Address> WorkAddresses { get; set; }
+            public List<Address> WorkAddresses { get; set; }

```

Comment:
Just to see results :) The optimization applies only for concrete types, the same as the compiler does it.